### PR TITLE
local changes on image data model

### PR DESCRIPTION
**Pull request by kordjamshidi**
Lets check if we have all recent changes before merging this PR

### CODE REVIEW

**Comment by Rahgooy** [comment]
It seems to be OK

**Comment by kordjamshidi** [comment]
why there are so many changes? are these the ones from Saul master only? plus the additional files from Umar?

**Comment by Rahgooy** [comment]
We didn't merge our changes to the master yet. They are all in `TaherMaster`.

**Comment by kordjamshidi** [comment]
okay, I see

File: data/mSprl/features.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+112	1	0.4417593         0.775     0.8583333     0.5560012     0.6382412      0.656913     0.7427811    0.01397768     0.3359084      119.3438      111.6017      105.8651      67.57657      64.81628      66.64162     0.3635511     0.5000695     0.7093432      68.94983      1.267164      3.321504      17.48109      8.993122      8.798025    -0.0448975     0.1654533      1.945854	120
```
Comment (Rahgooy) [inline data/mSprl/features.txt:1]: Why we have files from data folder?! it is ignored by git. isn't it?

File: saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/BaseTypes/Image.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+public class Image {
+    private final String label;
+    private String id;
+
+    public Image(String label, String id) {
+        this.label = label;
+        this.id = id;
+    }
+
+    public String getImageID()
```
Comment (Rahgooy) [inline saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/BaseTypes/Image.java]: I think it's better to change to `getId`

File: saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/BaseTypes/Image.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        return id;
+    }
+
+    public String getLabel()
+    {
+        return label;
+    }
+
+    public void setImageID(String ID)
```
Comment (Rahgooy) [inline saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/BaseTypes/Image.java]: `setId`

File: saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/BaseTypes/SegmentRelation.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
+package edu.illinois.cs.cogcomp.saulexamples.nlp.BaseTypes;
+
+/**
+ * Created by Umar Manzoor on 02/01/2017.
+ */
+public class SegmentRelation {
+    private String imageID;
+    private int segment_ID1;
+    private int segment_ID2;
```
Comment (Rahgooy) [inline saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/BaseTypes/SegmentRelation.java]: It's rare to use underscores for naming in java community it's more like C-Style, so I prefer `segmentId1` . If there are two segments, you can name them `firstSegmentId` and `secondSegmentId`

File: saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/BaseTypes/SegmentRelation.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package edu.illinois.cs.cogcomp.saulexamples.nlp.BaseTypes;
+
+/**
+ * Created by Umar Manzoor on 02/01/2017.
+ */
+public class SegmentRelation {
+    private String imageID;
```
Comment (Rahgooy) [inline saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/BaseTypes/SegmentRelation.java]: `imageId` if you follow camel conventions for naming

File: saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/BaseTypes/SegmentRelation.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        segment_ID2 = sg_id2;
+        relation = rel;
+    }
+
+    public String getRelation()
+    {
+        return relation;
+    }
+
+    public int getSegment_ID1()
```
Comment (Rahgooy) [inline saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/BaseTypes/SegmentRelation.java]: `getFirstSegmentId`

**Comment by kordjamshidi** [comment]
@Rahgooy thanks for your comments, but now a problem, how Umar can send fixes to this PR? :P

File: saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/BaseTypes/Segment.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * Created by Umar Manzoor on 29/12/2016.
+ */
+public class Segment {
+    private int segmentID;
+    private int segmentCode;
+    private String segmentFeatures;
+    private String segmentConcept;
+    private String imageID;
+    public Segment(String pID, int sID, int sC, String sF, String sCon)
```
Comment (Rahgooy) [inline saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/BaseTypes/Segment.java:12]: `sC`, `sF`, `sCon`?
I prefer `segmentId`, `imageId`

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/ImageSensors.scala [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package edu.illinois.cs.cogcomp.saulexamples.mSpRL2017
+
+import edu.illinois.cs.cogcomp.saulexamples.nlp.BaseTypes.{ Image, Segment, SegmentRelation }
+
+object ImageSensors {
+
+  def image_segment_link(i: Image, s: Segment): Boolean = {
```
Comment (Rahgooy) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/ImageSensors.scala]: please don't use underscores. The convention we have used so far in other data models suggests names like `imageToSegment`

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/mSpRL2017DataModel.scala [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+import edu.illinois.cs.cogcomp.saul.datamodel.DataModel
+import edu.illinois.cs.cogcomp.saulexamples.mSpRL2017.ImageSensors._
+import edu.illinois.cs.cogcomp.saulexamples.nlp.BaseTypes.{ Image, Segment, SegmentRelation }
+
+import scala.io.Source
+
+object mSpRL2017DataModel extends DataModel {
+
+  val image = node[Image]
```
Comment (Rahgooy) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/mSpRL2017DataModel.scala]: `images`

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/mSpRL2017DataModel.scala [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  val relation = node[SegmentRelation]
+
+  val image_segment = edge(image, segment)
+  // Linking associated Segments with Images
+  image_segment.addSensor(image_segment_link _)
+
+  // Here we will create relationships between different segments
+  // I am planning to use Relation class
+
+  val relToSg = edge(relation, segment)
```
Comment (Rahgooy) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/mSpRL2017DataModel.scala]: `relationToSegment` or `relationsToSegments`

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/mSpRL2017DataModel.scala [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+  relToSg.addSensor(rel_segment _)
+
+  val image_lable = property(image) {
```
Comment (Rahgooy) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/mSpRL2017DataModel.scala]: `imageLabel`

**Comment by Rahgooy** [comment]
@kordjamshidi 
I don't know. Maybe you should do that!

**Comment by kordjamshidi** [comment]
Ok, done. Please merge if this is ok.

**Pull request merged**